### ISSUE TITLE
extensions: api command does not process required query parameters.

### ISSUE THREAD

**Issue by reubenmiller** [reubenmiller/go-c8y-cli#254]
The `.required` property under a query parameter is not honoured.

**Example**

The following api command does not through an error even though the `dateFrom` and `dateTo` parameters (under the `.queryParameters` section are marked as required.

```sh
c8y oee supportedseries list --device Docker\ ASM\ Simulator\ Device
```

```yaml
# yaml-language-server: $schema=https://raw.githubusercontent.com/reubenmiller/go-c8y-cli/feat/extensions-manager/tools/schema/extensionCommands.json
---
group:
  name: supportedseries
  description: OEE supported series

commands:
  - name: list
    description: Get all supported series for the device
    descriptionLong: This service will take the deviceId as input and return the all supported series descriptors.
    method: GET
    path: service/oee-bundle/supportedseries/{device}
    exampleList:
      - command: c8y oee supportedseries list --device 12345 --dateFrom 1996-03-01T22:52:50.493Z --dateTo 1996-03-01T23:52:50.493Z
        description: Get support series for device with id 12345 within period from 1996-03-01T22:52:50.493Z to 1996-03-01T23:52:50.493Z
    pathParameters:
      - name: device
        type: device[]
        description: Device
        pipeline: true
        required: true
    queryParameters:
      - name: dateFrom
        type: datetime
        description: The beginning time to check the availability of series support
        required: true
      - name: dateTo
        type: datetime
        description: The end time to check the availability of series support
        required: true

```

**Workaround**

If there are required query parameters then they can be defined in the `.pathParameters` section instead, and add a reference to the variables.

```yaml
# yaml-language-server: $schema=https://raw.githubusercontent.com/reubenmiller/go-c8y-cli/feat/extensions-manager/tools/schema/extensionCommands.json
---
group:
  name: supportedseries
  description: OEE supported series

commands:
  - name: list
    description: Get all supported series for the device
    descriptionLong: This service will take the deviceId as input and return the all supported series descriptors.
    method: GET
    path: service/oee-bundle/supportedseries/{device}?dateFrom={dateFrom}&dateTo={dateTo}
    exampleList:
      - command: c8y oee supportedseries list --device 12345 --dateFrom 1996-03-01T22:52:50.493Z --dateTo 1996-03-01T23:52:50.493Z
        description: Get support series for device with id 12345 within period from 1996-03-01T22:52:50.493Z to 1996-03-01T23:52:50.493Z
    pathParameters:
      - name: device
        type: device[]
        description: Device
        pipeline: true
        required: true
      - name: dateFrom
        type: datetime
        description: The beginning time to check the availability of series support
        required: true
      - name: dateTo
        type: datetime
        description: The end time to check the availability of series support
        required: true
```

Afterwards the required parameters/flags are enforced.

```sh
c8y oee supportedseries list --device Docker\ ASM\ Simulator\ Device
2023-05-31T09:58:46.086+0200    ERROR   commandError: missing required parameters. [dateFrom dateTo]
```

**Comment by reubenmiller** [comment]
Fixed in v2.31.0

**Comment by reubenmiller** [comment]
Though it is still recommended just use sensible defaults for dateTo/dateFrom query parameters if the microservice requires them...it makes it much easier for users to use rather than having to provide the same dateTo/dateFrom each time.
